admin_ui/backend/api/calls.py

```
@@ -229,8 +240,47 @@ def _normalize_phase_tool_calls(entries: list, phase: str) -> list:
+    installs) just yields an empty map, so call-history responses still serve with
+    agent_name=None. Build this once per request and pass it into the converters to
+    avoid N+1 lookups."""
+    try:
+        from agents_store import AgentsStore
+        with AgentsStore() as store:  # close the sqlite connection promptly
+            return {
+                a["slug"]: a.get("display_name")
+                for a in store.list_all()
+                if a.get("slug")
+            }
+    except Exception:
```

Ruff (0.15.17)

[warning] Do not catch blind exception: `Exception`

(BLE001)